react-components/src/components/NodeCacheProvider/AssetMappingCacheProvider.tsx

```diff
@@ -0,0 +1,123 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { type ReactElement, type ReactNode, createContext, useContext, useMemo } from 'react';
+import { type CadModelOptions } from '../Reveal3DResources/types';
+import { AssetMappingCache, type NodeAssetMappingResult } from './AssetMappingCache';
+import { type UseQueryResult, useQuery } from '@tanstack/react-query';
+import { type AssetMapping3D } from '@cognite/sdk';
+import { useSDK } from '../RevealContainer/SDKProvider';
+import { useRevealKeepAlive } from '../RevealKeepAlive/RevealKeepAliveContext';
+import { type ModelId, type RevisionId, type TreeIndex } from './types';
+import { fetchAncestorNodesForTreeIndex } from './requests';
+
+export type AssetMappingCacheContent = {
+  cache: AssetMappingCache;
+};
+
+export type ModelWithAssetMappings = {
+  model: CadModelOptions;
+  assetMappings: AssetMapping3D[];
+};
+
+const AssetMappingCacheContext = createContext<AssetMappingCacheContent | undefined>(undefined);
+
+const useAssetMappingCache = (): AssetMappingCache => {
+  const content = useContext(AssetMappingCacheContext);
+
+  if (content === undefined) {
+    throw Error('Must use useAssetMappingCache inside a AssetMappingCacheContext');
+  }
+
+  return content.cache;
+};
+
+export const useMappeNodesForRevisions = (
```

Comment:
Should we add `asset` word to the name of the hook for easier distinction between this one and fdm based?

Reply:
Yes, good call